receiver/cloudfoundryreceiver/receiver_test.go

```diff
@@ -62,3 +69,651 @@ func TestDefaultValidLogsReceiver(t *testing.T) {
+	resource := metrics.ResourceMetrics().AppendEmpty()
+	resource.Resource().Attributes().PutStr("org.cloudfoundry.origin", "rep")
+
+	resource2 := pmetric.NewResourceMetrics()
+	resource2.Resource().Attributes().PutStr("org.cloudfoundry.origin", "rep2")
+
+	tests := []struct {
+		name          string
+		resourceAttrs bool
+		envelope      loggregator_v2.Envelope
+		metrics       pmetric.Metrics
+		expected      pmetric.ResourceMetrics
+	}{
+		{
+			name:          "resource attributes true",
+			envelope:      loggregator_v2.Envelope{},
+			resourceAttrs: false,
+			metrics:       pmetric.NewMetrics(),
+			expected:      pmetric.NewResourceMetrics(),
+		},
+		{
+			name:          "empty metrics",
+			envelope:      loggregator_v2.Envelope{},
+			resourceAttrs: true,
+			metrics:       pmetric.NewMetrics(),
+			expected:      pmetric.NewResourceMetrics(),
+		},
+		{
+			name: "matched resource metrics",
+			envelope: loggregator_v2.Envelope{
+				Tags: map[string]string{
+					"origin": "rep",
+					"custom": "datapoint",
+				},
+			},
+			resourceAttrs: true,
+			metrics:       metrics,
+			expected:      resource,
+		},
+		{
+			name: "non-matched resource metrics",
+			envelope: loggregator_v2.Envelope{
+				Tags: map[string]string{
+					"origin": "rep2",
+					"custom": "datapoint",
+				},
+			},
+			resourceAttrs: true,
+			metrics:       metrics,
+			expected:      resource2,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.resourceAttrs {
+				require.NoError(t, featuregate.GlobalRegistry().Set(allowResourceAttributes.ID(), true))
+				t.Cleanup(func() {
+					require.NoError(t, featuregate.GlobalRegistry().Set(allowResourceAttributes.ID(), false))
+				})
+			}
+			require.Equal(t, tt.expected, getResourceMetrics(tt.metrics, &tt.envelope))
+		})
+	}
+}
+
+func TestSetupLogsScope(t *testing.T) {
+	resourceSetup := plog.NewResourceLogs()
+	scope := resourceSetup.ScopeLogs().AppendEmpty()
+	scope.Scope().SetName(metadata.ScopeName)
+
+	tests := []struct {
+		name string
+		in   plog.ResourceLogs
+		out  plog.ResourceLogs
+	}{
+		{
+			name: "scope empty",
+			in:   plog.NewResourceLogs(),
+			out:  resourceSetup,
+		},
+		{
+			name: "scope set",
+			in:   resourceSetup,
+			out:  resourceSetup,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			setupLogsScope(tt.in)
+			require.Equal(t, tt.out, tt.in)
+		})
+	}
+}
+
+func TestGetResourceLogs(t *testing.T) {
+	logs := plog.NewLogs()
+	resource := logs.ResourceLogs().AppendEmpty()
+	resource.Resource().Attributes().PutStr("org.cloudfoundry.origin", "rep")
+
+	resource2 := plog.NewResourceLogs()
+	resource2.Resource().Attributes().PutStr("org.cloudfoundry.origin", "rep2")
+
+	tests := []struct {
+		name          string
+		resourceAttrs bool
+		envelope      loggregator_v2.Envelope
+		logs          plog.Logs
+		expected      plog.ResourceLogs
+	}{
+		{
+			name:          "resource attributes true",
```

Comment:
Name does not match the flag resourceAttrs

Reply:
adapted